packages/flutter/lib/src/cupertino/modal_bottom_sheet.dart

```diff
@@ -0,0 +1,389 @@
+class CupertinoScaffold extends StatefulWidget {
+  static _CupertinoScaffold? of(BuildContext context) =>
+      context.dependOnInheritedWidgetOfExactType<_CupertinoScaffold>();
+
+  final Widget body;
+  final Radius topRadius;
+  final Color transitionBackgroundColor;
+
+  const CupertinoScaffold({
+    Key? key,
+    required this.body,
+    this.topRadius = _default_top_radius,
+    this.transitionBackgroundColor = CupertinoColors.black,
+  }) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _CupertinoScaffoldState();
+
+  static Future<T> showCupertinoModalBottomSheet<T>({
+    required BuildContext context,
+    double? closeProgressThreshold,
+    required WidgetBuilder builder,
+    Curve? animationCurve,
+    Curve? previousRouteAnimationCurve,
+    Color? backgroundColor,
+    Color? barrierColor,
+    bool expand = false,
+    bool useRootNavigator = false,
+    bool bounce = true,
+    bool? isDismissible,
+    bool enableDrag = true,
+    Duration? duration,
+    RouteSettings? settings,
+    String? barrierLabel,
+  }) async {
+    assert(context != null);
+    assert(builder != null);
+    assert(expand != null);
+    assert(useRootNavigator != null);
+    assert(enableDrag != null);
+    assert(debugCheckHasMediaQuery(context));
+    
+    final scaffold = CupertinoScaffold.of(context)!;
+    final topRadius = scaffold.topRadius;
+    final result = await Navigator.of(context, rootNavigator: useRootNavigator)!
+        .push(CupertinoModalBottomSheetRoute<T>(
+      closeProgressThreshold: closeProgressThreshold,
+      builder: builder,
+      secondAnimationController: scaffold.animation,
+      containerBuilder: (_, __, Widget child) => _CupertinoBottomSheetContainer(
+        child: child,
+        backgroundColor: backgroundColor,
+        topRadius: topRadius,
+      ),
+      expanded: expand,
+      barrierLabel: barrierLabel,
+      bounce: bounce,
+      isDismissible: isDismissible ?? !expand,
+      modalBarrierColor:
+          barrierColor ?? CupertinoColors.black.withOpacity(0.12),
+      enableDrag: enableDrag,
+      topRadius: topRadius,
+      animationCurve: animationCurve,
+      previousRouteAnimationCurve: previousRouteAnimationCurve,
+      duration: duration,
+      settings: settings,
+    ));
+    return result;
+  }
+}
+
+class _CupertinoScaffoldState extends State<CupertinoScaffold>
+    with TickerProviderStateMixin {
+  late AnimationController animationController;
+
+  SystemUiOverlayStyle? lastStyle;
+
+  @override
+  void initState() {
+    animationController = AnimationController(
+        duration: const Duration(milliseconds: 350), vsync: this);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _CupertinoScaffold(
+      animation: animationController,
+      topRadius: widget.topRadius,
+      child: _CupertinoModalTransition(
```

Comment:
Not very clear what this group of classes is meant to solve. Could use some further description.

Reply:
I deleted it. People were having problems changing the default MaterialPageRoute to MaterialWithModalPageRoute when using the package, so I created this widget `CupertinoScaffold` that would animate the same without needed to change the route. I think it is not needed anymore as the default routes would support the animation without any change needed